pkg/controller/labelidentity/label_group_index_test.go

```diff
@@ -0,0 +1,613 @@
+		labelMatch := newLabelIdentityMatch(tt.label, 1)
+		matched := labelMatch.matches(tt.selector)
+		if tt.expectMatch != matched {
+			t.Errorf("Unexpected matching status for %s and %s. Exp: %v, Act: %v", tt.label, tt.selector.getKey(), tt.expectMatch, matched)
+		}
+	}
+}
+
+func TestAddSelector(t *testing.T) {
+	tests := []struct {
+		name                 string
+		selectorToAdd        *types.GroupSelector
+		existingPolicyKey    string
+		policyKey            string
+		expMatchedIDs        []uint32
+		expLabelIdentityKeys sets.String
+		expPolicyKeys        sets.String
+	}{
+		{
+			name:                 "cluster-wide app=web",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+			policyKey:            "policyA",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyA"),
+		},
+		{
+			name:                 "cluster-wide app=web another policy",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nil, nil, nil),
+			existingPolicyKey:    "policyA",
+			policyKey:            "policyB",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyA", "policyB"),
+		},
+		{
+			name:                 "pod app=web and ns purpose=test",
+			selectorToAdd:        types.NewGroupSelector("", pSelWeb, nsSelTest, nil, nil),
+			policyKey:            "policyB",
+			expMatchedIDs:        []uint32{1},
+			expLabelIdentityKeys: sets.NewString(labelA),
+			expPolicyKeys:        sets.NewString("policyB"),
+		},
+		{
+			name:                 "cluster-wide app=db",
+			selectorToAdd:        types.NewGroupSelector("", pSelDB, nil, nil, nil),
+			policyKey:            "policyC",
+			expMatchedIDs:        []uint32{2, 3},
+			expLabelIdentityKeys: sets.NewString(labelB, labelC),
+			expPolicyKeys:        sets.NewString("policyC"),
+		},
+		{
+			name:                 "app=db in ns testing",
+			selectorToAdd:        types.NewGroupSelector("testing", pSelDB, nil, nil, nil),
+			policyKey:            "policyD",
+			expMatchedIDs:        []uint32{2},
+			expLabelIdentityKeys: sets.NewString(labelB),
+			expPolicyKeys:        sets.NewString("policyD"),
+		},
+		{
+			name:                 "app=db in ns random",
+			selectorToAdd:        types.NewGroupSelector("random", pSelDB, nil, nil, nil),
+			policyKey:            "policyE",
+			expMatchedIDs:        []uint32{},
+			expLabelIdentityKeys: sets.NewString(),
+			expPolicyKeys:        sets.NewString("policyE"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			i := NewLabelIdentityIndex()
+			i.labelIdentities = map[string]*labelIdentityMatch{
+				labelA: labelMatchA,
+				labelB: labelMatchB,
+				labelC: labelMatchC,
+			}
+			i.labelIdentityNamespaceIndex = map[string]sets.String{
+				"testing": sets.NewString(labelA, labelB),
+				"nomatch": sets.NewString(labelC),
+			}
```

Comment:
By https://github.com/antrea-io/antrea/pull/3913#discussion_r1018970322, I meant using the methods to prepare the test data, instead of operating the cache directly. For example,
```
i.AddLabelIdentity(labelA, 1)
i.AddLabelIdentity(labelB, 2)
i.AddLabelIdentity(labelC, 3)
```
The above code could acheive the same and it's more readable and more close to what happens when running the code in actual env.

Reply:
Sorry it seems that I missed addressing that comment.. Done